### PR TITLE
Move the DDSQL docs out to it's own section

### PR DESCRIPTION
**Pull request by estherk15**
### What does this PR do? What is the motivation?

- DDSQL product is not functionally part of Dashboards, though in the app it's listed under Dashboards.
- PM requested to move it out to its own section
- Images were left alone, will move out in a separate PR
- [DOCS-8626](https://datadoghq.atlassian.net/browse/DOCS-8626)

### Merge instructions

- [x] Please merge after reviewing

### Additional notes

[DOCS-8626]: https://datadoghq.atlassian.net/browse/DOCS-8626?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: content/en/ddsql_editor/guide/_index.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+disable_toc: true
+cascade:
+    algolia:
+        rank: 20
+        category: Guide
+        subcategory: Graphing Guides
+---
+
+{{< whatsnext desc="General Guides:" >}}
+    {{< nextlink href="ddsql_editor/guide/ddsql_use_cases" >}}Common DDSQL queries and use cases{{< /nextlink >}}
```
Comment (hestonhoffman) [inline content/en/ddsql_editor/guide/_index.md:13]: Currently 404ing, but the fix above should work

File: content/en/dashboards/guide/ddsql_use_cases.md
Comment (hestonhoffman) [inline content/en/dashboards/guide/ddsql_use_cases.md:1]: I think this one needs to be moved to ./ddsql_editor/guides

Also needs an alias if you end up moving it

**Review by hestonhoffman — APPROVED** [review]
Looks good - just one broken link

File: content/en/dashboards/guide/ddsql_use_cases.md
Comment (estherk15) [reply content/en/dashboards/guide/ddsql_use_cases.md:1]: Thanks for catching this!

**Pull request merged**